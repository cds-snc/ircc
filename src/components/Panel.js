import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Container from './Container'
import Header from './Header'

const Content = styled.div`padding: 0 0.8em;`

const Panel = ({ title, width, children }) =>
  <Container width={width}>
    <Header title={title} />
    <Content>
      {children}
    </Content>
  </Container>

Panel.propTypes = {
  title: PropTypes.string,
  width: PropTypes.string,
  children: PropTypes.object,
}

export default Panel
