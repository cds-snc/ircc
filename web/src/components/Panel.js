import React from 'react'
import PropTypes from 'prop-types'
import styled from 'react-emotion'
import Container from './Container'
import Header from './Header'

const Content = styled.div`padding: 0 0.8em;`

const Panel = ({ title, width, children }) =>
  <Container width={width}>
    <Header title={title} />
    <Content>
      {children}
    </Content>
  </Container>

Panel.propTypes = {
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  width: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
}

export default Panel
