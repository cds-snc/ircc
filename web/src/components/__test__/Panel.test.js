import React from 'react'
import { mount, shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import 'jest-styled-components'
import Panel from '../Panel'

describe('<Panel />', () => {
  it('can be rendered', () => {
    let wrapper = shallow(<Panel title="How to pay your fees" />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('displays the title', () => {
    let text = 'How to pay your fees'
    let wrapper = mount(<Panel title={text} width="30em" />)
    expect(wrapper.text()).toEqual(text)
  })
})
