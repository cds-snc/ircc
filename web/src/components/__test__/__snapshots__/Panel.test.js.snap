// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel /> can be rendered 1`] = `
<styled.div>
  <Header
    title="How to pay your fees"
  />
  <styled.div />
</styled.div>
`;
